pkg/generate/app/dockerimagelookup.go

```diff
@@ -108,6 +112,13 @@ func (r DockerClientSearcher) Search(terms ...string) (ComponentMatches, error)
 		}
 
 		componentMatches = append(componentMatches, termMatches...)
+		if len(componentMatches) == 0 && r.AllowMissing {
```

Comment:
The searcher will also be used when you do a call with --search=true. I suppose in that case we wouldn't want the --allow-missing flag to be true. Maybe we should add validation to the flags in the command to make them mutually exclusive.